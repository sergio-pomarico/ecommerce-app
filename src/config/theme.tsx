import React, {ReactNode} from 'react';
import {
  createTheme,
  useTheme as useReTheme,
  ThemeProvider as ReStyleThemeProvider,
} from '@shopify/restyle';
import {
  ImageStyle,
  TextStyle,
  ViewStyle,
  FlexStyle,
  Dimensions,
} from 'react-native';

type NamedStyles<T> = {
  [P in keyof T]: ViewStyle | TextStyle | ImageStyle | FlexStyle;
};

const {width} = Dimensions.get('screen');
export const aspectRatio = width / 375;

const palette = {
  black: '#000000',
  blue: '#7DCCEC',
  lightBlue: '#D3F2FF',
  danger: '#E16C71',
  grey: '#9A9A9D',
  darkGrey: '#868686',
  lightGrey: '#F5F5F8',
  mediumGrey: '#C4C4C4',
  indigo: '#5956E9',
  ligthIndigo: '#F4F6FA',
  red: '#FA4A0C',
  white: '#FFFFFF',
};

const theme = createTheme({
  colors: {
    background: palette.lightGrey,
    primary: palette.indigo,
    text: palette.black,
    white: palette.white,
    blue: palette.blue,
    grey: palette.grey,
    darkGrey: palette.darkGrey,
    error: palette.danger,
  },
  spacing: {
    xs: 8,
    s: 16,
    m: 24,
    l: 32,
    xl: 64,
  },
  borderRadii: {
    s: 4,
    m: 10,
    l: 25,
    xl: 64,
  },
  breakpoints: {},
  textVariants: {
    hero: {
      color: 'white',
      fontSize: 50,
      fontFamily: 'Raleway-ExtraBold',
      lineHeight: 50,
      textAlign: 'center',
    },
    h1: {
      fontFamily: 'Raleway-SemiBold',
      fontSize: 32,
      lineHeight: 36,
      textAlign: 'center',
    },
    h2: {
      fontFamily: 'Raleway-Bold',
      fontSize: 28,
      lineHeight: 32,
      textAlign: 'center',
    },
    h3: {
      fontFamily: 'Raleway-SemiBold',
      fontSize: 24,
      lineHeight: 24,
      textAlign: 'center',
    },
    product_card_title: {
      fontFamily: 'Raleway-Regular',
      fontSize: 16,
      fontWeight: '600',
      lineHeight: 18,
    },
    price: {
      fontFamily: 'Raleway-Regular',
      fontSize: 15,
      fontWeight: '600',
      lineHeight: 17,
    },
    button_large: {
      fontFamily: 'Raleway-Bold',
      fontSize: 20,
      lineHeight: 24,
      textAlign: 'center',
    },
    button_medium: {
      fontFamily: 'Raleway-Bold',
      fontSize: 16,
      lineHeight: 18,
      textAlign: 'center',
    },
    label: {
      fontFamily: 'Raleway-SemiBold',
      fontSize: 15,
      lineHeight: 18,
      textTransform: 'capitalize',
    },
  },
});

export const ThemeProvider = ({children}: {children: ReactNode}) => (
  <ReStyleThemeProvider {...{theme}}>{children}</ReStyleThemeProvider>
);

export type Theme = typeof theme;
export const useTheme = () => useReTheme<Theme>();

export const makeStyle =
  <T extends NamedStyles<T>>(styles: (theme: Theme) => T) =>
  () => {
    const curretTheme = useTheme();
    return styles(curretTheme);
  };

export default theme;
