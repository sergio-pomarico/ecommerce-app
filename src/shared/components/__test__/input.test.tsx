import React from 'react';
import {fireEvent, render} from '@testing-library/react-native';
import {ThemeProvider} from '@config/theme';
import {Input} from '@components';

describe('Input test suite', () => {
  const testID = 'input';
  const props = {
    onChanceText: jest.fn(),
    onBlur: jest.fn(),
    touched: false,
    error: '',
    label: 'Email',
    placeholder: 'Add your email',
    testID,
  };
  test('check if the Input render correctly', () => {
    const {toJSON} = render(
      <ThemeProvider>
        <Input {...props} />
      </ThemeProvider>,
    );
    expect(toJSON()).toMatchSnapshot();
  });
  test('forwards all props to the Input', () => {
    const {getByTestId} = render(
      <ThemeProvider>
        <Input {...props} />
      </ThemeProvider>,
    );

    const input = getByTestId(testID);

    fireEvent.changeText(input, 'testing!');

    expect(props.onChanceText).toHaveBeenCalled();
    expect(props.onChanceText).toHaveBeenCalledWith('testing!');
  });
});
