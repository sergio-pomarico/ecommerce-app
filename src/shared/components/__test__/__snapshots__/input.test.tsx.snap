// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input test suite check if the Input render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "height": 120,
        "justifyContent": "flex-start",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "flex-start",
          "borderBottomColor": "#9A9A9D",
          "borderBottomWidth": 0.5,
          "marginBottom": 8,
          "paddingBottom": 16,
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-end",
            "flexDirection": "row",
            "justifyContent": "flex-end",
            "marginBottom": 16,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#868686",
              "fontFamily": "Raleway-SemiBold",
              "fontSize": 15,
              "lineHeight": 18,
              "marginLeft": 8,
              "textTransform": "capitalize",
            },
          ]
        }
      >
        Email
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
        ]
      }
    >
      <TextInput
        autoCapitalize="none"
        error=""
        onBlur={[MockFunction]}
        onChangeText={[Function]}
        placeholder="Add your email"
        style={
          Object {
            "flex": 1,
          }
        }
        testID="input"
        touched={false}
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
</View>
`;
